app/ddm/auth/page.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import * as React from 'react';
+import { checkAuth, checkAuthFormat } from '@data/api';
+
+import styles from './page.module.scss';
+
+import Button from '@components/Button';
+import Input from '@components/basic/Input';
+import { useRouter, useSearchParams } from 'next/navigation';
+import { loadAuth, saveAuth, saveDDMAddress } from '@root/common/ddm';
+
+export default function Auth(props: {}) {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
+  // Store a tmp auth token in the component so the main application auth token
+  // state doesn't update until the user submits the form
+  const [tmpAuthToken, setTmpAuthToken] = React.useState(loadAuth() || '');
+  const [tmpDDMAddress, setTmpDDMAddress] = React.useState(loadAuth() || process.env.NEXT_PUBLIC_API_URL || 'http://localhost:1415');
+  const [loading, setLoading] = React.useState(false);
+
+  async function onSubmit(e) {
```

Comment:
It's a little impure to have this onSubmit function get its declaration within the closure of its parent, you should expand onSubmit to take a few more arguments and bring it outside of this closure.

Reply:
i split this function into pieces and moved it out of the component body